portal-impl/src/com/liferay/portal/security/permission/AdvancedPermissionChecker.java

```diff
@@ -1559,6 +1574,7 @@ private boolean _isGroupAdminImpl(Group group) throws Exception {
 	private static final Log _log = LogFactoryUtil.getLog(
 		AdvancedPermissionChecker.class);
 
+	private Map<Long, long[]> _contributedRoleIds;
```

Comment:
This is wrong, please revert.

We can use one permission checker to test permissions on several different `groupId`s . With this change we would mix the granted roles and introduce a security vulnerability. See for example here: https://github.com/liferay/liferay-portal/blob/8a6b3d6bba6f63258524dc172c2ed0341ba7b703/portal-impl/src/com/liferay/portal/service/impl/UserServiceImpl.java#L2377

Side note - usually permission checkers are scoped to the request but it's because they are stored inside PermissionThreadLocal. In general, as Ray pointed out yesterday, we allow to create bunch of permission checkers for different userIds in one request. Just heads up if you consider using TL cache then we need userId to be part of the key.

Reply:
Oh, yes, you are right, somehow overlooked it. Then it's fine, sorry for the inconvenience.